### PR TITLE
Added ExeptionHandler for HabitController to solve IllegalArgumentsExeption

### PR DESCRIPTION
**Pull request by VinuVicho**
# GreenCityS PR

## Issue Link :clipboard:
https://github.com/ita-social-projects/GreenCity/issues/6561
https://github.com/ita-social-projects/GreenCity/issues/6562

## Summary Of Changes :fire:
Added ExeptionHandler to handle situations, when invalid parameter is entered into sort and language field

## Added
ExeptionHandler method in HabitController 

## How to test :clipboard:
Repeat actions in folowing requests: https://github.com/ita-social-projects/GreenCity/issues/6561 https://github.com/ita-social-projects/GreenCity/issues/6562

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-6561 6562)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells or duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: core/src/main/java/greencity/controller/HabitController.java [hunk 1]
```diff
@@ -34,15 +34,7 @@
...
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RequestPart;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import org.springframework.web.bind.annotation.PutMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/HabitController.java]: please do not use `*` import

File: core/src/main/java/greencity/controller/HabitController.java [hunk 2]
```diff
@@ -285,4 +277,14 @@ public ResponseEntity<CustomHabitDtoResponse> updateCustomHabit(@PathVariable Lo
         return ResponseEntity.status(HttpStatus.OK)
             .body(habitService.updateCustomHabit(request, habitId, principal.getName(), image));
     }
+
+    /**
+     * Handles IllegalArgumentException.
+     *
+     * @return Error message "Bad Request".
+     */
+    @ExceptionHandler({IllegalArgumentException.class})
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/HabitController.java]: Can we add an exception handler for IllegalArgumentException here https://github.com/ita-social-projects/GreenCity/blob/dev/core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java ?

In this case, we will cover the same exception for other endpoints

File: core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java [hunk 3]
```diff
@@ -177,6 +178,19 @@ public final ResponseEntity<Object> handleConstraintViolationException(Constrain
...
     }
 
+    /**
+     * Handles IllegalArgumentException.
+     *
+     * @return Error message "Bad Request".
+     * @author Volodymyr Mladonov
+     */
+    @ExceptionHandler({IllegalArgumentException.class})
+    public ResponseEntity<Object> handleException(IllegalArgumentException ex, WebRequest request) {
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java]: please change the method name to `handleIllegalArgumentException`

File: core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java [hunk 4]
```diff
@@ -5,6 +5,7 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import greencity.constant.AppConstant;
 import greencity.constant.ErrorMessage;
+import greencity.constant.HttpStatuses;
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java]: unused import

File: core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java [hunk 5]
```diff
@@ -177,6 +178,19 @@ public final ResponseEntity<Object> handleConstraintViolationException(Constrain
         return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(exceptionResponse);
     }
 
+    /**
+     * Handles IllegalArgumentException.
+     *
+     * @return Error message "Bad Request".
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java]: This method does not return `message "Bad Request"`
it returns `HttpStatus.BAD_REQUEST` and the response body will depend on the input params

**Comment by RomaMocherniuk** [comment]
I see that we do not have unit tests for `CustomExceptionHandler` class. Have you tested it locally?

**Comment by VinuVicho** [comment]
> I see that we do not have unit tests for `CustomExceptionHandler` class. Have you tested it locally?

I have tested the CustomExceptionHandler class locally. When using an invalid input locally, we receive a response of 400 instead of the previous 500 error.

**Review by RomaMocherniuk — APPROVED** [review]

**Review by HelenSotnik — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Review by Yurii-Midianyi — APPROVED** [review]

**Pull request merged**